activejdbc/src/main/java/org/javalite/activejdbc/Model.java

```diff
@@ -249,6 +261,79 @@ protected void hydrate(Map<String, Object> attributesMap, boolean fireAfterLoad)
+        return changedAttributeNames;
+    }
+
+    /**
+     * Verifies if the passed value for attributeName would set this instance to modified state.
+     * @param attributeName
+     * @param newValue
+     * @return
+     */
+    protected boolean wouldAttributeValueModifyModel(String attributeName, Object newValue) {
+
+        Object currentAttributeValue = get(attributeName);
+
+        if (isNew() || (currentAttributeValue == null && newValue != null))
+            return true;
+
+        if (currentAttributeValue != null) {
+
+            if (newValue == null)
+                return true;
+
+            if (newValue.getClass() != currentAttributeValue.getClass()) {
+                Object convertedObject = convertValueToTargetClass(newValue, currentAttributeValue);
+                if (convertedObject != null) {
+                    return !convertedObject.equals(currentAttributeValue);
+                } else
+                    return true;
+            } else {
+                return !newValue.equals(currentAttributeValue);
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Tries to convert the given new attribute value type to the one of the existing attribute value.
+     * @param newAttributeValue
+     * @param existingAttributeValue
+     * @return
+     */
+    protected Object convertValueToTargetClass(Object newAttributeValue, Object existingAttributeValue) {
+
+        Object result = null;
+
+        try {
+            if (existingAttributeValue instanceof java.sql.Timestamp) {
+                result = Convert.toTimestamp(newAttributeValue);
+            } else if (existingAttributeValue instanceof java.sql.Time) {
+                result = Convert.toTime(newAttributeValue);
+            } else if (existingAttributeValue instanceof java.sql.Date) {
+                result = Convert.toSqlDate(newAttributeValue);
+            } else if (existingAttributeValue instanceof Long) {
+                result = Convert.toLong(newAttributeValue);
+            } else if (existingAttributeValue instanceof Integer) {
+                result = Convert.toInteger(newAttributeValue);
+            } else if (existingAttributeValue instanceof Short) {
+                result = Convert.toShort(newAttributeValue);
+            } else if (existingAttributeValue instanceof BigDecimal) {
+                result = Convert.toBigDecimal(newAttributeValue);
+            } else if (existingAttributeValue instanceof Boolean) {
+                result = Convert.toBoolean(newAttributeValue);
+            } else if (existingAttributeValue instanceof Double) {
+                result = Convert.toDouble(newAttributeValue);
+            } else if (existingAttributeValue instanceof Float) {
+                result = Convert.toFloat(newAttributeValue);
+            } else if (existingAttributeValue instanceof String) {
+                result = Convert.toString(newAttributeValue);
+            }
+
+        } catch(Exception e) {}
```

Comment:
Shouldn't there be some sort of `logging` here?

Reply:
Sure, that's possible and makes sense, since some of the convert methods themselves can throw an exception.

Reply:
should not catch exception, it (exception) needs to propagate to next level. Just logging is masking a problem.